runtime/src/bank/check_transactions.rs

```diff
@@ -71,6 +79,10 @@ impl Bank {
         let hash_queue = self.blockhash_queue.read().unwrap();
         let last_blockhash = hash_queue.last_hash();
         let next_durable_nonce = DurableNonce::from_blockhash(&last_blockhash);
+        // safe so long as the BlockhashQueue is consistent
```

Comment:
this is definitely the case on mnb. is it on testnet and devnet as well?

Reply:
if it is we should probably, separately, remove all the `FeeRateGovernor` stuff that could cause this assumption to be incorrect.

Reply:
I'm not really sure what you're specifically referring to about the `FeeRateGovernor` here (though I do agree it should be removed). This safety comment was copied from `fn last_blockhash_and_lamports_per_signature` which was committed a few years ago.

Reply:
> Maybe I also misunderstood what was meant by "consistent" here?

I think it just means that we can safely unwrap because if there was a last hash, there must be a lps value for that hash.

> BlockhashQueue's lamports_per_signature is populated from data that ultimately is controlled by FeeRateGovernor. If that FeeRateGovernor is configured to actually adjust lamports_per_signature then it'd make the BlockhashQueue inconsistent

What you say is true, the blockhash queue's lps is indeed populated by the fee rate governor which is still configured to adjust lps and so different hashes can have different lps values. Assuming this PR makes changes correctly to match existing behavior, all is fine right? Or is there something else you're getting at?

Reply:
Nope, think its safe. Was a misunderstanding on my part - resolved!